### PR TITLE
Python: Update minimum required version to 3.9, check version at runtime

### PR DESCRIPTION
**Pull request by jbemmel**
Triggered by https://github.com/ipspace/netlab/issues/1689

### CODE REVIEW

**Comment by ssasso** [comment]
Ubuntu 20.04 has python 3.8 by default.

Either we force the install script to install python3.9 (available also for 20.04), or we remove also Ubuntu 20.04 from the supported ubuntu versions.

**Comment by ipspace** [comment]
> Either we force the install script to install python3.9 (available also for 20.04), or we remove also Ubuntu 20.04 from the supported ubuntu versions.

We'll go for the third option and keep writing code that still works on Python 3.9 ;)

File: netsim/__init__.py [hunk 1]
```diff
@@ -6,14 +6,18 @@
...
 
+if sys.version_info < (3,9):
+  print(f"FATAL ERROR: Netlab requires Python version 3.9 or newer - installed {sys.version}")
+  abort = True
+
 try:
   import box
 
-  if box.__version__ < '7.2.0':                 # netlab needs Python Box version 7.0 or higher
+  if box.__version__ < '7.2.0':                 # netlab needs Python Box version 7.2 or higher
```
Comment (ipspace) [inline netsim/__init__.py:16]: And you chose to change the minimum Box version as well. I wonder what the justification was.

Actually, I don't want to know.

Comment (jbemmel) [reply netsim/__init__.py:16]: I did not change the version, I corrected the comment to match the code

**Pull request closed without merging**